### PR TITLE
MVP for Activity Feed

### PR DESCRIPTION
**Pull request by liana193**
- added a new class: Variables that groups together the classes of User, Messages, and Conversations
- Used a sorted list to display activities that occur

### CODE REVIEW

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 1]
```diff
@@ -11,16 +10,77 @@
...
+
+	 /**
+	  * Set up state for handling login-related requests. This method is only called when running in a
+	  * server, not when running in a test.
+	 */
+	@Override
+	public void init() throws ServletException {
+       super.init();
+       setConversationStore(ConversationStore.getInstance());
+       setMessageStore(MessageStore.getInstance());
```
Comment (kmontgomery) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: small nit: as per the style guide, these lines should be indented like line 44!

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 2]
```diff
@@ -11,16 +10,77 @@
...
+	  
+	/**
+	 * Sets the UserStore used by this servlet. This function provides a common setup method for use
+	 * by the test framework or the servlet's init() function.
+	 */
+	void setUserStore(UserStore userStore) {
+	  this.userStore = userStore;
+	}
+
+	/**
```
Comment (kmontgomery) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: small nit: this javadoc isn't formatted properly.

File: src/main/java/codeu/controller/ActivityFeedServlet.java [hunk 3]
```diff
@@ -11,16 +10,77 @@
...
+     * This function fires when a user requests the /activityfeed URL. It simply forwards the request to
+	 * activityfeed.jsp.
+	 */
    @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
+   public void doGet(HttpServletRequest request, HttpServletResponse response)
```
Comment (kmontgomery) [inline src/main/java/codeu/controller/ActivityFeedServlet.java]: small nit: the only change on this line is adding an unneeded space. please revert!

File: src/main/webapp/WEB-INF/view/activityfeed.jsp [hunk 4]
```diff
@@ -16,15 +43,67 @@
...
         <h2 style="color:red"><%= request.getAttribute("error") %></h2>
     <% } %>
- 	<p> This is the Activity Feed </p>
+ 	<p>	Here's everything that's happened on the site so far!</p>
+      
+    <div id="activityfeed">  
+      <ul>
+    <%
+ 	ArrayList<Activity> events = new ArrayList<Activity>();
+ 	events.addAll(PersistentStorageAgent.getInstance().loadUsers());
```
Comment (kmontgomery) [inline src/main/webapp/WEB-INF/view/activityfeed.jsp]: Rather than getting a new instance each time, it would be more efficient to create a variable for PersistentStorageAgent.getInstance() and reuse it on each of these lines.

File: src/main/webapp/WEB-INF/view/activityfeed.jsp [hunk 5]
```diff
@@ -16,15 +43,67 @@
...
+ 	events.addAll(PersistentStorageAgent.getInstance().loadConversations());
+ 	events.addAll(PersistentStorageAgent.getInstance().loadMessages());
+ 	
+ 	Collections.sort(events, new Comparator<Activity>() {
```
Comment (kmontgomery) [inline src/main/webapp/WEB-INF/view/activityfeed.jsp]: please fix the whitespace formatting of this file as per the style guide.

File: src/test/java/codeu/controller/ActivityFeedServletTest.java [hunk 6]
```diff
@@ -22,21 +26,23 @@
   private HttpServletRequest mockRequest;
   private HttpServletResponse mockResponse;
   private RequestDispatcher mockRequestDispatcher;
-
+  
   @Before
-  public void setup() throws IOException {
+  public void setup() throws IOException, ServletException {
```
Comment (kmontgomery) [inline src/test/java/codeu/controller/ActivityFeedServletTest.java]: Why did you need to add this?

File: src/main/webapp/WEB-INF/view/activityfeed.jsp (hunk 4 above)
Comment (liana193) [reply src/main/webapp/WEB-INF/view/activityfeed.jsp]: Done.

File: src/test/java/codeu/controller/ActivityFeedServletTest.java (hunk 6 above)
Comment (liana193) [reply src/test/java/codeu/controller/ActivityFeedServletTest.java]: It was showing up as an error on eclipse

File: src/main/webapp/WEB-INF/view/activityfeed.jsp (hunk 5 above)
Comment (liana193) [reply src/main/webapp/WEB-INF/view/activityfeed.jsp]: Done.

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 3 above)
Comment (liana193) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: Done.

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 2 above)
Comment (liana193) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: Done.

File: src/main/java/codeu/controller/ActivityFeedServlet.java (hunk 1 above)
Comment (liana193) [reply src/main/java/codeu/controller/ActivityFeedServlet.java]: Done.
